### DISCUSSION TITLE
Use `border-radius: 10%` for "state-badge" elements

### DISCUSSION THREAD

**Discussion by ildar170975** [home-assistant/frontend#13140] [Dashboard]
The `state-badge` element is used to display an icon in some Lovelace cards.

Mainly icons have a transparent background.
Currently by default the `state-badge` element has a `border-radius: 50%` property.
It becomes visible if you apply some background-color  to the `state-badge`:
![изображение](https://user-images.githubusercontent.com/71872483/177835767-ad876dc9-ebd2-4351-a044-989ecf1208c2.png)

Imagine that some entity has an `entity_picture` property specified.
Then we will see a picture in a circle:
![изображение](https://user-images.githubusercontent.com/71872483/177835998-b9a3e5f3-5dbb-4f6b-8471-f41dc9dd501d.png)

For person's pictures this could be acceptable.
But what if an image represents some other stuff - an actual device, a flora etc?
Surely I may apply a `border-radius: 10%` style to the `state-badge`:
![изображение](https://user-images.githubusercontent.com/71872483/177837374-ec972364-e480-43a9-bd96-b27ecd63844c.png)

I suggest to use that `border-radius: 10%` style as a default style for ALL `state-badge` elements.
I propose to start with Entities card & Logbook card.

**Comment by KTibow** [comment]
Right now, Home Assistant uses rounded entity pictures in most places. If you want to change it to a rounded square, I'd recommend using card-mod.
Why do you want to use a rounded square instead? Could you give some reasons to change away from the current system, which is widely used?

**Reply by ildar170975** [reply]
1. Regarding card-mod. Surely I may specify a custom card-mod theme to make images of square shape.
But then I will have to specify a style for many places, like:
- entiity-row
- glance card
- more-info window
- logbook
etc.

OK, this may be cumbersome but it is possible.
But - is some cases card-mod simply does not work.
For example: due to recent changes in Logbook card it works differently; if the card is placed into some other card (like `config-template-card`) the style does not stick.
Also, even earlier I myself found impossible to style entity-rows inside `restriction-card`, 'config-template-card' - see issues on a card-mod Github. Also, a possibility to style elements in `auto-entities` & some elements in `fold-entity-row` is also broken for long time. So, card-mod may not help in some cases.


2. Most of pictures are not supposed to be used as "circular".
If a picture represents an object of rectangular shape - I will have to reduce it's size, otherwise it will not fit a circular border.
The same story happened a few years ago when Apple decided to show contacts' avatars as circular: original photos in most cases are not supposed to be used in a circle, the are of rectangular shape. Unfortunately, other companies started using circular contacts too and call this "a modern style" (((

3. Making a "placeholder" for an mdi-icon of square shape will not cause any damage for icons, these icons have a transparent background, it will affect images only.

**Reply by KTibow** [reply]
Interesting reasons. Although right now, many places are designed with circles in mind, eg chips, badges, the map.
![image](https://user-images.githubusercontent.com/10727862/179024121-98703d98-d7e9-4d0f-83bc-b7754e5483b6.png)
![image](https://user-images.githubusercontent.com/10727862/179024373-87adce0b-d2ac-4fde-b941-1418013d18e0.png)
The only place where I would want to see more of the picture would be in camera previews or icons, eg:
![image](https://user-images.githubusercontent.com/10727862/179025025-5c8b7222-df70-4ca8-8c2b-18ac0c2ec5a0.png)
Human faces usually fit well into circular spaces - none of your face is cropped off in the screenshot you sent.

**Reply by ildar170975** [reply]
> Although right now, many places are designed with circles in mind, eg chips, badges, the map.

In most places people use MDI icons - and you cannot see any differences since (as I said before) icons have a transparent background.
In your particular case with your personal avatar - if it is originally of a circular shape - you will not loose it since the circle is not clipped if placed into a square border.




> The only place where I would want to see more of the picture would be in camera previews or icons, eg:

Yes, good example.
And what about rectangular images of physical objects? (equipment, trees, plants, ....)
I do not want them to be clipped or scaled down to fit into circular border.



> Human faces usually fit well into circular spaces - none of your face is cropped off in the screenshot you sent.

Yes, because I chose this photo from many others to use as an avatar.
But due to this STUPID CIRCULAR MODERN STYLE people cannot use ANY of their pictures.
What if I would like a photo from my sport competitions? Surely I was not posing for a picture and it could be just impossible to fit the image into a circular shape.

This war started a few years ago as I said when Apple introduced these circular photos.
Originally a person makes a photo using a special cropping  - specially to add some details, background etc.
And all these efforts are just trashed out due to some "modern style".
Try to create a circular avatar of a group of people (family, for instance), a laying pet etc.

**Reply by KTibow** [reply]
Good point. Maybe you could move this to the forum, where more people can see it.

**Reply by ildar170975** [reply]
Probably you are right.
Thank you very much for a support!
Created this topic:
https://community.home-assistant.io/t/suggestion-round-images-used-in-frontend-should-be-square/440117

**Comment by ildar170975** [comment]
Meanwhile a square shape was made for state-badge for `media_player` entities:
https://github.com/home-assistant/frontend/pull/18066
![271680972-ac134e75-b522-4507-90c1-3695e9309a37](https://github.com/home-assistant/frontend/assets/71872483/c2a1ad7e-f0fa-4542-9cef-8bbc99d0be35)

Thinking about making the same for all entities with `entity_picture`.
